test/e2e/app-dir/hooks/hooks.test.ts

```diff
@@ -54,12 +54,26 @@ createNextDescribe(
     describe('useSearchParams', () => {
       it('should have the correct search params', async () => {
         const $ = await next.render$(
-          '/hooks/use-search-params?first=value&second=other%20value&third'
+          '/hooks/use-search-params/dynamic?first=value&second=other%20value&third'
         )
         expect($('#params-first').text()).toBe('value')
         expect($('#params-second').text()).toBe('other value')
         expect($('#params-third').text()).toBe('')
         expect($('#params-not-real').text()).toBe('N/A')
+
+        // dynamic page doesn't have bail out
+        expect($('html#__next_error__').length).toBe(0)
+        expect($('meta[content=noindex]').length).toBe(0)
+      })
+
+      it('should not contain noindex meta tag for static built page', async () => {
```

Comment:
wdyt about moving this into a separate test? I worry that as we add new unrelated assertions the test apps themselves become bloated (like app-static or parallel-routes-and-interception) so it's harder to understand what's being tested and we may accidentally change behavior of another unrelated test.